packages/xml-views-validation/src/validators/elements/unknown-tag-name.ts

```diff
@@ -0,0 +1,199 @@
+  model: UI5SemanticModel
+): UnknownTagNameIssue[] {
+  // Can't give an error if there is no position
+  if (xmlElement.syntax.openName == undefined || xmlElement.name === null) {
+    return [];
+  }
+
+  // To avoid false positives, we only validate tags with recognized namespaces (from the model)
+  // or without a namespace (and for tags without a namespace, we only validate if
+  // they cannot be user-defined classes or aggregations).
+
+  // Get the namespace
+  if (xmlElement.ns !== undefined) {
+    // The cast is necessary because typescript doesn't recognize the conditions above as type guards
+    return validateTagWithNamespace(xmlElement as NamedXMLElementWithNS, model);
+  } else {
+    // The cast is necessary because typescript doesn't recognize the condition above as a type guard
+    return validateTagWithoutNamespace(xmlElement as NamedXMLElement, model);
+  }
+}
+
+type NamedXMLElement = XMLElement & {
+  name: string;
+  syntax: { openName: XMLToken };
+};
+type NamedXMLElementWithNS = NamedXMLElement & { ns: string };
+
+function validateTagWithNamespace(
+  xmlElement: NamedXMLElementWithNS,
+  model: UI5SemanticModel
+): UnknownTagNameIssue[] {
+  const issueDefaults = {
+    kind: "UnknownTagName" as "UnknownTagName",
+    offsetRange: {
+      start: xmlElement.syntax.openName.startOffset,
+      end: xmlElement.syntax.openName.endOffset,
+    },
+    severity: "error" as "error",
+  };
+
+  // To avoid false positives, we assume unrecognized namespaces are user-defined so we don't validate tags in them
+  // (they could be user-defined class tags).
+  // They could also be valid special namespaces like xhtml.
+  // TODO There should be an error in xml-tools if the namespace is not defined in an xmlns attribute
+  const { namespace: ui5Namespace } = getUI5NodeFromXMLElementNamespace(
+    xmlElement,
+    model
+  );
+  if (ui5Namespace === undefined) {
+    return [];
+  }
+
+  // Check if it's a known class (aggregations don't have namespaces)
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
+  if (ui5Class !== undefined) {
+    return [];
+  }
+
+  return [
+    {
+      ...issueDefaults,
+      message: getMessage(
+        UNKNOWN_CLASS_IN_NS,
+        xmlElement.name,
+        ui5NodeToFQN(ui5Namespace)
+      ),
+    },
+  ];
+}
+
+function validateTagWithoutNamespace(
+  xmlElement: NamedXMLElement,
+  model: UI5SemanticModel
+): UnknownTagNameIssue[] {
+  const issueDefaults = {
+    kind: "UnknownTagName" as "UnknownTagName",
+    offsetRange: {
+      start: xmlElement.syntax.openName.startOffset,
+      end: xmlElement.syntax.openName.endOffset,
+    },
+    severity: "error" as "error",
+  };
+
+  // If the default namespace is defined and it's not a known namespace, don't validate
+  // (it could be a custom class tag or a special namespace like xhtml or template)
+  const {
+    namespace: ui5Namespace,
+    isXmlnsDefined,
+  } = getUI5NodeFromXMLElementNamespace(xmlElement, model);
```

Comment:
`getUI5NodeFromXMLElementNamespace` is confusing for me.
I don't see that we actually need the UI5Node, we can bulid the FQN from the XML AST.
Should we have a simpler utility or set of utilities that check if:
- Extract the XML AstNode namespace and baseName parts (if this does not already exist).
- A matching UI5Namespace **exists** in the model?

Reply:
Would this improve readability?